lib/publiccloud/ec2.pm

```diff
@@ -253,4 +253,21 @@ sub start_instance
     $instance->public_ip($public_ip);
 }
 
+sub get_instance_type
+{
+    my ($self, $instance, %args) = @_;
+    my $instance_id = $instance->instance_id();
+    my $i_desc = $self->describe_instance($instance);
+    return $i_desc->{InstanceType};
+}
+
+sub change_instance_type
+{
+    my ($self, $instance, $instance_type, %args) = @_;
+    die "Instance type is already $instance_type" if ($self->get_instance_type($instance) eq $instance_type);
+    my $instance_id = $instance->instance_id();
+    assert_script_run("aws ec2 modify-instance-attribute --instance-id $instance_id --instance-type '{\"Value\": \"$instance_type\"}'", quiet => 1);
```

Comment:
is this call synchronous or not ? we sure that after command returns it is always immediately affect describe instance call ? maybe we need to have at least several attempts before dying ?